tests/useful_mocks.py

```diff
@@ -5,27 +5,48 @@
 from unittest_protocols import Ability, Action, Question
 
 
-def get_mock_action(**kwargs):
+def get_mock_action(**kwargs) -> mock.Mock:
     mock_action = mock.create_autospec(Debug(), _name="fakeaction", instance=True)
     mock_action.perform_as = mock.create_autospec(Debug().perform_as, **kwargs)
     mock_action.describe.return_value = "An African or a European swallow?"
     return mock_action
 
 
-def get_mock_question():
-    return mock.create_autospec(Question, instance=True)
+def get_mock_question() -> mock.Mock:
+    question = mock.create_autospec(Question, instance=True)
+    question.describe.return_value = "Description of Mocked Question"
+    question.answered_by.return_value = True
+    return question
 
 
-def get_mock_resolution():
+def get_mock_resolution() -> mock.Mock:
     return mock.create_autospec(BaseResolution, instance=True)
 
 
-def get_mock_task():
+def get_mock_task() -> mock.Mock:
     """Get a describable mock task."""
     task = mock.create_autospec(Action, instance=True)
     task.describe.return_value = "A mocked task."
     return task
 
 
-def get_mock_ability():
+def get_mock_ability() -> mock.Mock:
     return mock.create_autospec(Ability, instance=True)
+
+
+class FakeAbility:
+    def forget(self):
+        return
+
+
+class AnotherFakeAbility:
+    def forget(self):
+        return
+
+
+class FakeQuestion:
+    def answered_by(self, the_actor) -> bool:
+        return True
+
+    def describe(self) -> str:
+        return "Fake question"
```

Comment:
Same question with this class while we have `get_mock_question` above.

Reply:
For some reason I hesitated using the mock here, likely just self doubt.  The tests pass just fine if I use the mock instead of `FakeQuestion`.  I'll update this.

Reply:
I remember now why I was using a FakeQuestion class.  It was to get around the fact that mock objects don't trigger usage errors in mypy.  Mypy sees mock as basically `Any`.   Part of the reason for this PR was to verify our tests were making proper calls (spoiler: there were some incorrect ones).   Unfortunately I don't have a better solution to this. We would either need to 

- forego all mypy checking and use mocks nearly everywhere
- have a mix of mock and "real" objects to allow for SOME mypy checking.

Neither solution is great.